tree/dataframe/src/RLoopManager.cxx

```diff
@@ -529,6 +529,8 @@ void RLoopManager::CleanUpTask(unsigned int slot)
 /// This method also clears the contents of GetCodeToJit().
 void RLoopManager::Jit()
 {
+   R__LOCKGUARD(gROOTMutex);
```

Comment:
can we reduce the locking to just the `std::move(GetCodeToJit())`?

Reply:
I don't think we can remove it because then we could assume that code is already jitted although it's not yet available. I see crashes in the tests along these lines.

Reply:
So if i put the guard only round the move assignment.

Reply:
Does this make sense?

Reply:
yes you are completely right